meson.build

```diff
@@ -1,10 +1,12 @@
 project('spdlog', 'cpp', version : '0.16.3', license: 'MIT',
-		default_options : ['cpp_std=c++11'])
+        default_options : ['cpp_std=c++11'])
 
 inc = include_directories('include')
 
 thread_dep = dependency('threads')
 
-subdir('tests')
+if not meson.is_subproject()
```

Comment:
I disagree with this approach. But I have alternatives that might work for you.
1. Instead of `meson.is_subproject()` you can use `get_option('tests')` and create `meson_options.txt` [according to the guide](http://mesonbuild.com/Build-options.html).
2. Fix `tests/meson.build` by introducing `-WnoWARNING` flags. Check them with `has_argument()` of [compiler object](http://mesonbuild.com/Reference-manual.html#compiler-object) for compatibility with other compilers.
3. Use `is_system` to suppress header warnings, e.g. [gtest](https://github.com/mesonbuild/gtest/blob/1.8.0/googlemock/meson.build). It will still issue warnings for the initial build, but it will not trigger warnings on each rebuild of your own code.

Reply:
1 sounds like a good idea. How would I pass the option to `subproject('spdlog')` then?

Reply:
See example: https://github.com/mesonbuild/zlib/tree/1.2.11

It will lead to
```meson
Project options:
  Option    Current Value Possible Values Description                              
  ------    ------------- --------------- -----------                              
  zlib:solo false         [true, false]   Build a minimal zlib without file I/O API
```

Reply:
How can projects set that value when using zlib?